### DISCUSSION TITLE
[Feature]: How can I get current Route'handle

### DISCUSSION THREAD

**Discussion by guaijie** [remix-run/react-router#9812] [Q&A]
### What is the new or updated feature that you are suggesting?

useMatches only works with a data router like [createBrowserRouter](https://reactrouter.com/en/main/routers/create-browser-router), since they know the full route tree up front and can provide all of the current matches. Additionally, useMatches will not match down into any descendant route trees since the router isn't aware of the descendant routes.

So, How can I get handles for any descendant route

So, I expect useMatch's return value to contain both the current handle and the parent handle

### Why should this feature be included?
This lets me get rid of the useMatches to deal with the breadcrumb problem, because instead of getting the descendant route, I'm getting the current and parent route data, which I know for sure

const {  handles } = useMatch()

If you do not want to change the useMatch api, you can create a new hook useHandles

const  handles = useHandles()

The least you can do is give me another way to get the handle on the descendant route

**Comment by mazerty** [comment]
it's actually a problem even to get the handle for the current route... right now i have to loop through the result of useMatches, find the one matching the current location, then extract the handle from there :(

`useMatches().find(i => i.pathname === useLocation().pathname).handle`

**Comment by dxiongya** [comment]
matchRoutes(routes, pathname) ?

**Reply by mazerty** [reply]
this doesn't return the handle unfortunately
